### PR TITLE
chore: java sample ui updates for data pipelines

### PR DESCRIPTION
**Pull request by mrehan27**
part of [MBL-299](https://linear.app/customerio/issue/MBL-299/update-sample-app-to-display-new-config-values)

### Changes

- Update Java Layout UI to use new naming convention for DataPipelines
- Updated variable naming to match new implementation

### CODE REVIEW

File: samples/resources/values/appium.xml [hunk 1]
```diff
@@ -10,12 +10,12 @@
...
-    <string name="acd_cdp_api_key_input">CDP API Key Input</string>
+    <string name="acd_api_host_input">Track URL Input</string>
+    <string name="acd_cdn_host_input">CDN Host Input</string>
+    <string name="acd_cdp_api_key_input">API Key Input</string>
     <string name="acd_site_id_input">Site ID Input</string>
-    <string name="acd_api_key_input">API Key Input</string>
-    <string name="acd_bq_seconds_delay_input">BQ Seconds Delay Input</string>
-    <string name="acd_bq_min_tasks_input">BQ Min Number of Tasks Input</string>
+    <string name="acd_flush_interval_input">BQ Seconds Delay Input</string>
+    <string name="acd_flush_at_input">BQ Min Number of Tasks Input</string>
```
Comment (mrehan27) [inline samples/resources/values/appium.xml:18]: I didn't change the values to avoid breaking it unnecessarily for Appium.

**Comment by mrehan27** [comment]
Changed required for Kotlin sample have been done in separate PR #337

File: samples/java_layout/src/main/java/io/customer/android/sample/java_layout/sdk/CustomerIORepository.java [hunk 2]
```diff
@@ -51,64 +58,40 @@ public void initializeSdk(SampleApplication application) {
...
+
+        final String apiHost = sdkConfig.getApiHost();
+        if (!TextUtils.isEmpty(apiHost)) {
+            builder.setApiHost(apiHost);
+        }
+
+        final String cdnHost = sdkConfig.getCdnHost();
+        if (!TextUtils.isEmpty(cdnHost)) {
+            builder.setCdnHost(cdnHost);
         }
```
Comment (levibostian) [inline samples/java_layout/src/main/java/io/customer/android/sample/java_layout/sdk/CustomerIORepository.java:79]: Same as https://github.com/customerio/customerio-android/pull/337#discussion_r1603966585

File: samples/java_layout/src/main/java/io/customer/android/sample/java_layout/ui/settings/SettingsActivity.java [hunk 3]
```diff
@@ -140,16 +136,25 @@ private void setupObservers() {
...
-        // Since SDK does not allow tracking URL with empty host or incorrect schemes
-        return !TextUtils.isEmpty(uri.getAuthority()) && ("http".equals(scheme) || "https".equals(scheme)) && uri.getPath().endsWith("/");
+        try {
+            Uri uri = Uri.parse(url);
+            // Since SDK does not support custom schemes, we manually append http:// to the URL
+            // So the URL is considered invalid if it ends with a slash, contains a scheme, query or fragment
+            return url.endsWith("/")
+                    || !TextUtils.isEmpty(uri.getScheme())
+                    || !TextUtils.isEmpty(uri.getQuery())
+                    || !TextUtils.isEmpty(uri.getFragment());
```
Comment (levibostian) [inline samples/java_layout/src/main/java/io/customer/android/sample/java_layout/ui/settings/SettingsActivity.java:152]: Same as https://github.com/customerio/customerio-android/pull/337#discussion_r1603970406

**Review by levibostian — APPROVED** [review]

File: samples/java_layout/src/main/java/io/customer/android/sample/java_layout/sdk/CustomerIORepository.java (hunk 2 above)
Comment (mrehan27) [reply samples/java_layout/src/main/java/io/customer/android/sample/java_layout/sdk/CustomerIORepository.java:79]: Replied in linked comment, we only allow non-null values here today.

File: samples/java_layout/src/main/java/io/customer/android/sample/java_layout/ui/settings/SettingsActivity.java (hunk 3 above)
Comment (mrehan27) [reply samples/java_layout/src/main/java/io/customer/android/sample/java_layout/ui/settings/SettingsActivity.java:152]: Replied in linked comment.

**Pull request merged**